clarifai/datasets/upload/multimodal.py

```diff
@@ -0,0 +1,66 @@
+import uuid
+from concurrent.futures import ThreadPoolExecutor
+from typing import List, Tuple, Type
+
+from clarifai_grpc.grpc.api import resources_pb2
+from google.protobuf.struct_pb2 import Struct
+
+from clarifai.client.input import Inputs
+from clarifai.datasets.upload.base import ClarifaiDataLoader, ClarifaiDataset
+
+
+class MultiModalDataset(ClarifaiDataset):
+
+  def __init__(self, data_generator: Type[ClarifaiDataLoader], dataset_id: str) -> None:
+    super().__init__(data_generator, dataset_id)
+
+  def _extract_protos(
+      self,
+      batch_input_ids: List[str],
+  ) -> Tuple[List[resources_pb2.Input]]:
+    """ Creats Multimodal (image and text) input protos for batch of input ids.
+        Args:
+            batch_input_ids: List of input IDs to retrieve the protos for.
+        Returns:
+            input_protos: List of input protos.
+
+        """
+    input_protos, annotation_protos = [], []
+
+    def process_data_item(id):
+      data_item = self.data_generator[id]
+      metadata = Struct()
+      image_bytes = data_item.image_bytes
+      text = data_item.text
+      labels = data_item.labels if isinstance(data_item.labels, list) else [data_item.labels]
+      input_id = f"{self.dataset_id}-{uuid.uuid4().hex[:8]}" if data_item.id is None else f"{self.dataset_id}-{str(data_item.id)}"
```

Comment:
`{uuid.uuid4().hex[:8]}` 

There can be util method from [utils](https://github.com/Clarifai/clarifai-python/tree/master/clarifai/utils) package that generates and returns a random hex string of size specified in input parameter e.g. 8 in this case